### DISCUSSION TITLE
Use of ignore_errors: true should be considered a linting error

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#343] [Ideas] [answered]
# Issue Type
- Feature request

# Desired Behaviour

When linter encounters use of `ignore_errors: true` it should consider an error because (almost) always this is an example of anti-pattern, which hides real errors, marks tasks as failed when in fact they are not failed (expected non zero return code), or they trigger ansible debugger for no reason.

The easiest way to avoid these is to do `failed_when: false` instead of `ignore_errors: true`, as this would not have the side-effects.

I wrote a more detailed article about this on https://medium.com/@sbarnea/why-ansible-ignore-errors-is-evil-500fb6e81229 

Also note that earlier this week I convinced Ansible team to implement an option that would allow disabling the debugger triggering when ignore_errors is true, but this feature will be include din Ansible 2.7. Also this future feature would address only the debugger part, all the others are not addressed.

<img width="513" alt="ansible-ignore-errors-side-effects" src="https://user-images.githubusercontent.com/102495/39861544-d50a220a-5438-11e8-9799-a5acb6aad915.png">

**Comment by sugar700** [comment, marked as answer]
I believe this is now implemented (https://github.com/ansible/ansible-lint/pull/1540).

**Reply by xorinzor** [reply]
I don't think this is something that should have been changed.

A valid use case for example is when validation steps are performed. If the user configures a property that some resource will be recreated on errors, you'd want to still see the errors in the log, but continue execution regardless.

Changing `failed_when` instead would no longer mark these as errors, whereas they are (and still should be) regardless of whether they are ignored or not.
